UKGovernmentBEIS/inspect_ai
# Add `SandboxEnvironmentConfig` and `SandboxEnvironmentConfigType`

To allow sandbox environment providers to accept specialised config types - not just `str | None`.

For instance, a K8s sandbox environment provider might want to allow customisation of: chart (either a path to a local dir, or a chart from a repository), chart version, and path to values (maybe even a dict). Whilst this could be done with colon-separated strings, it soon gets messy and hard to support backward compatibility.

## Manual testing

I've run an eval from the command line `inspect eval examples/tool_use.py --sandbox docker:browser/compose.yaml` to make sure passing a config str from the CLI still works.

## Usage example

```py
Task(
    ...,
    sandbox=SandboxEnvironmentSpec(
        "podman", PodmanSandboxEnvironmentConfig(socket="/path/to/socket")
    ),
)
```

## Caveats

When a string config is passed, we do some path normalisation (I think making it relative to cwd for example). This won't be done with paths inside a `SandboxEnvironmentConfig`.
